storage/src/system/auction/detail.rs

```diff
@@ -152,14 +156,16 @@ impl ValidatorBidsDetail {
         validator_public_key: &PublicKey,
         era_ending: EraId,
         include_credit: bool,
+        cap: Ratio<U512>,
     ) -> U512 {
         if !include_credit {
             return U512::zero();
         }
 
         if let Some(inner) = self.validator_credits.get(validator_public_key) {
             if let Some(credit) = inner.get(&era_ending) {
-                return credit.amount();
+                let amount = Ratio::new_raw(credit.amount(), U512::one());
```

Comment:
@EdHastingsCasperAssociation this appears to be throttling the credit rather than capping it. What I meant is capping applied credit by a percentage of your actual CSPR stake. E.g., if I have 100 CSPR staked, and consume 200 CSPR worth of gas, my weight in the era when the credit applies should not exceed 120 (with a cap of 1/5). But if I consume 10 worth of CSPR, it should be 110.

Reply:
Ah...ok, I'll make the adjustment.